app/src/main/java/org/mozilla/fenix/compose/NoCollectionsMessage.kt

```diff
@@ -0,0 +1,164 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.ext.dashedBorder
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * [NoCollectionsMessage] for displaying a message detailing the collections feature and
+ * allowing users to easily start creating their first.
+ *
+ * @param showAddToCollectionButton to check if there is at least one normal tab.
+ * @param onAddTabsToCollectionClick {lambda} to create a new first collection.
+ * @param onRemoveClick {lambda} to remove empty collections message.
+ */
+@Composable
+fun NoCollectionsMessage(
+    showAddToCollectionButton: Boolean,
+    onAddTabsToCollectionClick: () -> Unit,
+    onRemoveClick: () -> Unit,
+) {
+    Box(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {}
+            .dashedBorder(
+                color = FirefoxTheme.colors.borderPrimary,
+                cornerRadius = 8.dp,
+                dashHeight = 2.dp,
+                dashWidth = 4.dp
+            )
+            .padding(
+                start = 16.dp,
+                bottom = 16.dp
+            )
+    ) {
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                SectionHeader(
+                    text = stringResource(R.string.collections_header),
+                    modifier = Modifier
+                        .weight(1f)
+                        .wrapContentHeight(align = Alignment.Top)
+                )
+
+                IconButton(
+                    onClick = onRemoveClick
+                ) {
+                    Icon(
+                        painter = painterResource(R.drawable.ic_close),
+                        contentDescription = stringResource(
+                            R.string.remove_home_collection_placeholder_content_description
+                        ),
+                        modifier = Modifier.size(24.dp),
+                        tint = FirefoxTheme.colors.iconPrimary
+                    )
+                }
+            }
+
+            PrimaryText(
+                text = stringResource(R.string.no_collections_description2),
+                modifier = Modifier.fillMaxWidth(),
+                fontSize = 14.sp
+            )
+
+            if (showAddToCollectionButton) {
+                Spacer(modifier = Modifier.height(12.dp))
+
+                AddToCollectionButton(
+                    onClick = onAddTabsToCollectionClick
+                )
+            }
+        }
+    }
+}
+
+/**
+ * No Collections Tabs Button
+ *
+ * @param onClick The lambda called when the button is clicked.
```

Comment:
```suggestion
 * @param onClick Invoked when the user clicks on the button.
```

Reply:
See https://github.com/mozilla-mobile/fenix/pull/24530#discussion_r850561178 on why I am choosing to do it this way